fili-core/src/main/java/com/yahoo/bard/webservice/web/endpoints/JobsServlet.java

```diff
@@ -415,11 +423,12 @@ protected void handleResultsResponse(
      * return an Observable wrapping the PreResponse as is.
      *
      * @param preResponse  The PreResponse to be inspected
+     * @param jobsApiRequest  JobsApiRequest object with all the associated info with it
      *
      * @return An Observable wrapping the PreResponse or an Observable wrapping a ResponseException
      *
      */
-    protected Observable<PreResponse> handlePreResponseWithError(PreResponse preResponse) {
+    protected Observable<PreResponse> handlePreResponseWithError(PreResponse preResponse, ApiRequest jobsApiRequest) {
```

Comment:
Rather than taking in the entire `ApiRequest`, this method should only take the pieces that it needs (the `PaginationParameters` and the `UriInfo` I believe). We've been burned before by taking the entire `ApiRequest` when we only needed a couple pieces of it.